src/browser/pages/vscode.ts

```diff
@@ -5,18 +5,63 @@ const options = getOptions()
 // TODO: Add proper types.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-let nlsConfig: any
+// NOTE@jsjoeio
+// This lives here ../../../lib/vscode/src/vs/base/common/platform.ts#L106
+export const nlsConfigElementId = "vscode-remote-nls-configuration"
+
+type NlsConfiguration = {
+  locale: string
+  availableLanguages: { [key: string]: string } | {}
+  _languagePackId?: string
+  _translationsConfigFile?: string
+  _cacheRoot?: string
+  _resolvedLanguagePackCoreLocation?: string
+  _corruptedFile?: string
+  _languagePackSupport?: boolean
+  loadBundle?: any
+}
+
+/**
+ * A helper function to get the NLS Configuration settings.
+ *
+ * This is used by VSCode for localizations (i.e. changing
+ * the display language).
+ *
+ * Make sure to wrap this in a try/catch block when you call it.
+ **/
+export function getNlsConfiguration(document: Document) {
+  const errorMsgPrefix = "[vscode]"
+  const nlsConfigElement = document?.getElementById(nlsConfigElementId)
+  const nlsConfig = nlsConfigElement?.getAttribute("data-settings")
+
+  if (!document) {
+    throw new Error(`${errorMsgPrefix} Could not parse NLS configuration. document is undefined.`)
+  }
+
+  if (!nlsConfigElement) {
+    throw new Error(
+      `${errorMsgPrefix} Could not parse NLS configuration. Could not find nlsConfigElement with id: ${nlsConfigElementId}`,
+    )
+  }
+
+  if (!nlsConfig) {
+    return undefined
```

Comment:
Should this throw too? I think we can consider it an error.

Reply:
Oh! I thought the `nlsConfig` was optional but you're right, this should throw!

Reply:
Fixed!